packages/dom/src/dom/test/is-number-input.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * Internal dependencies
+ */
+import isNumberInput from '../is-number-input';
+
+describe( 'isNumberInput', () => {
+	it( 'should handle zero value properly', () => {
+		const input = document.createElement( 'input' );
+		input.type = 'number';
+		input.value = 0;
+		expect( isNumberInput( input ) ).toBe( true );
+	} );
+} );
```

Comment:
Totally up to you, but I don't think we gain much with this tautological test. :)